src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -34,26 +36,87 @@ export default class VizShonenJump extends Connector {
         let data = await this.fetchDOM(request);
 
         if ( !data.innerText.includes('Latest free chapters') ) {
             alert('This website is geolocked. It can only be accessed from the USA.\nYou may use MANGA Plus instead.', this.label, 'info');
             return;
         }
-
-        data = [...data.querySelectorAll('div.o_sort_container div > a')];
+        
+        data = [...data.querySelectorAll('a.o_chapters-link')];
 
         return data.map(manga => {
             return {
-                id: manga.pathname,
-                title: manga.querySelector(':nth-child(2)').innerText.trim()
+                id: JSON.stringify({
+                    id: manga.pathname,
+                    title: manga.innerText.trim(),
+                    entry_type: this.EntryTypeEnum.chapter
+                }),
+                title: manga.innerText.trim(),
             };
         });
     }
 
-    async _getChapters(manga) {
+    async _getMangas() {
+        const mangaAvailibleByVolumes = await this._getMangasAvailibleByVolumes();
+        var mangaAvailibleByChapters = await this._getMangasAvailibleByChapters();
+
+        mangaAvailibleByVolumes.forEach(mangaAvailibleByVolume => {
+                const index = mangaAvailibleByChapters.findIndex(element => element.title === mangaAvailibleByVolume.title);
+                // Found it.
+                if (index != -1) {
+                    var complexManga = JSON.parse(mangaAvailibleByChapters[index].id);
+
+                    mangaAvailibleByChapters[index] = {
+                        id: JSON.stringify({
+                            id: complexManga.id,
+                            volume_id: mangaAvailibleByVolume.volume_id,
+                            title: mangaAvailibleByVolume.title,
+                            entry_type: this.EntryTypeEnum.volume | this.EntryTypeEnum.chapter
+                        }),
+                        title: mangaAvailibleByVolume.title,
+                    };
+                }
+                // Didn't find it, just add this manga straight into the list.
+                else {
+                    mangaAvailibleByChapters.push({
+                        id: JSON.stringify({
+                            volume_id: mangaAvailibleByVolume.volume_id,
+                            id: mangaAvailibleByVolume.volume_id,
+                            title: mangaAvailibleByVolume.title,
+                            entry_type: this.EntryTypeEnum.volume
+                        }),
+                        title: mangaAvailibleByVolume.title,
+                    });
+                }
+            })
+
+        return mangaAvailibleByChapters;
+    }
+
+    async _getMangaChapters(manga) {
+        if ((manga.entry_type & this.EntryTypeEnum.chapter) != this.EntryTypeEnum.chapter) {
+            console.log(`Exiting from _getMangaChapters ${JSON.stringify(manga)}`);
```

Comment:
_getMangaVolumes?

Reply:
This was just for testing purposes, I'm just going to remove it. The stringification was from trying to get something more than like `[Object object]` to print out.